src/authentication/DPoPWebIdExtractor.ts

```diff
@@ -13,31 +13,38 @@ import { CredentialsExtractor } from './CredentialsExtractor';
 export class DPoPWebIdExtractor extends CredentialsExtractor {
   protected readonly logger = getLoggerFor(this);
   private readonly targetExtractor: TargetExtractor;
+  private readonly solidIdentityVerifier: VerifySolidIdentityFunction;
 
   public constructor(targetExtractor: TargetExtractor) {
     super();
     this.targetExtractor = targetExtractor;
+    this.solidIdentityVerifier = createSolidIdentityVerifier();
   }
 
   public async canHandle({ headers }: HttpRequest): Promise<void> {
-    const { authorization } = headers;
-    if (!authorization || !authorization.startsWith('DPoP ')) {
-      throw new NotImplementedHttpError('No DPoP Authorization header specified.');
+    const { authorization, dpop } = headers;
+    if (!authorization) {
+      throw new BadRequestHttpError('No authorization header specified.');
+    }
+    if (!authorization.startsWith('DPoP ')) {
+      throw new BadRequestHttpError('Authorization header is not DPoP bound.');
+    }
+    if (!dpop) {
+      throw new BadRequestHttpError('No DPoP header specified.');
```

Comment:
The original code is correct; we need to have the `dpop` error in the `handle` method.

The question the `canHandle` method is:
- Should I be the one handling this (as opposed to someone else)?
Clearly, if the `Authorization` starts with `DPoP`, the answer is "Yes, I should be handling this".

Then, only when actually handling this, in the `handle` method, it should say: "I am the one handling this, and you've made mistake".

Otherwise, if doing this in `canHandle` as above, you're saying "I am not the one handling this situation", so it will be flagged as "we need a different handler" rather than "this handler is authorized to say the client made a mistake".

Note that neither a missing authorization headers, nor the absence of `DPoP` therein, is a client _error_: it just means this handler is not the right one. But _if_ the `Authorization` header _does_ contain `DPoP`, then the absence of a `DPoP` header _is_ an error. (So no `BadRequestHttpError`, but rather "this handler does not implement it (another one will)").

Reply:
It all makes sense, thanks for the explanation!